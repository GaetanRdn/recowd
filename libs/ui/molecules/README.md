# ui-molecules

This library was generated with [Nx](https://nx.dev).

## description

Contains all components which are molecules. E.g. atom + functional meaning.
