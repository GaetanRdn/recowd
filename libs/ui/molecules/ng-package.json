{
  "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../dist/libs/ui/molecules",
  "lib": {
    "entryFile": "src/index.ts"
  }
}
