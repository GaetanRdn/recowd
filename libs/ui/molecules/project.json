{
  "projectType": "library",
  "sourceRoot": "libs/ui/molecules/src",
  "prefix": "rc",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:ng-packagr-lite",
      "outputs": ["dist/libs/ui/molecules"],
      "options": {
        "project": "libs/ui/molecules/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/ui/molecules/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/ui/molecules/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/ui/molecules"],
      "options": {
        "jestConfig": "libs/ui/molecules/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/ui/molecules/**/*.ts",
          "libs/ui/molecules/**/*.html"
        ]
      }
    }
  },
  "tags": ["ui", "molecules"]
}
