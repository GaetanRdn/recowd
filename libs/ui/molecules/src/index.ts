export * from './lib/avatars/rw-avatar/rw-avatar.component';

export * from './lib/buttons-icons/contact-button-icon/contact-button-icon.component';

export * from './lib/buttons-with-icons/education-button-with-icon/education-button-with-icon.component';
export * from './lib/buttons-with-icons/experience-button-with-icon/experience-button-with-icon.component';
export * from './lib/buttons-with-icons/partners-button-with-icon/partners-button-with-icon.component';

export * from './lib/cards/education-card/education-card.component';
export * from './lib/cards/work-in-progress-card/work-in-progress-card.component';

export * from './lib/custom-icons/angular-custom-icon/angular-custom-icon.component';
export * from './lib/custom-icons/cbp-custom-icon/cbp-custom-icon.component';
export * from './lib/custom-icons/cypress-custom-icon/cypress-custom-icon.component';
export * from './lib/custom-icons/jest-custom-icon/jest-custom-icon.component';
export * from './lib/custom-icons/norsys-custom-icon/norsys-custom-icon.component';
export * from './lib/custom-icons/nx-custom-icon/nx-custom-icon.component';
export * from './lib/custom-icons/sass-custom-icon/sass-custom-icon.component';
export * from './lib/custom-icons/storybook-custom-icon/storybook-custom-icon.component';

export * from './lib/lists/roles-list/roles-list.component';
export * from './lib/lists/technologies-list/technologies-list.component';

export * from './lib/partner-link/partner-link.component';
