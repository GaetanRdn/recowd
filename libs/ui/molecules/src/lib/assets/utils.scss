@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
