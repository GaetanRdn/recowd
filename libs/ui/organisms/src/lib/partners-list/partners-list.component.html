<rc-card *ngFor="let partner of partners">
  <rc-card-content>
    <rc-partner-link [partner]="partner"></rc-partner-link>
  </rc-card-content>
</rc-card>
