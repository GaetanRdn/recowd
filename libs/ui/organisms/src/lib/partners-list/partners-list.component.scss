:host {
  align-items: center;
  display: flex;
  flex-flow: row wrap;
  gap: 4rem;
  justify-content: center;

  rc-card {
    background-color: white;
    max-height: clamp(15rem, 15rem, 25rem);
    overflow: hidden;

    .rc-card-content {
      transform: translateY(-28%);
    }
  }
}
