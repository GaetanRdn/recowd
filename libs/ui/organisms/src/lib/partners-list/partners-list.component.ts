import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { PartnerLinkComponent } from '@recowd/ui/molecules';
import { Partner } from '@recowd/models';
import { CardModule } from '@recowd/ui-atoms';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'rc-partners-list',
  standalone: true,
  imports: [CommonModule, PartnerLinkComponent, CardModule],
  templateUrl: './partners-list.component.html',
  styleUrls: ['./partners-list.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PartnersListComponent {
  @Input() public partners: Partner[] = [];
}
