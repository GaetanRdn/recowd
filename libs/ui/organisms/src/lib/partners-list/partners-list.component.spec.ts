import { TestBed, waitForAsync } from '@angular/core/testing';

import { PartnersListComponent } from './partners-list.component';
import { TemplateLookup } from '@recowd/test/utils';
import { NorsysCustomIconComponent } from '@recowd/ui/molecules';
import { Component } from '@angular/core';

describe('PartnersListComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [PartnersListComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it('should create empty', () => {
    templateLookup.detectChanges();

    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  it('should create', () => {
    templateLookup.hostComponent.partners = [
      {
        link: 'www.paulo.com',
        icon: NorsysCustomIconComponent,
      },
    ];

    templateLookup.detectChanges();

    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-partners-list [partners]="partners"></rc-partners-list>`,
})
class HostComponent {
  public partners!: PartnersListComponent['partners'];
}
