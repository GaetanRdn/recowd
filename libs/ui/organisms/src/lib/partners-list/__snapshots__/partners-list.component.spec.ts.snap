// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnersListComponent should create 1`] = `
<rc-partners-list>
  <rc-card>
    <div
      class="rc-card-header"
    />
    <rc-card-content
      class="rc-card-content"
    >
      <rc-partner-link>
        <a
          class="rc-button-icon"
          href="www.paulo.com"
          rcbuttonicon=""
          target="_blank"
        >
          <rc-cbp-custom-icon>
            <rc-custom-icon>
              <img
                alt="cbp"
                height="300"
                src="assets/images/cbp.svg"
                width="300"
              />
            </rc-custom-icon>
          </rc-cbp-custom-icon>
          
        </a>
      </rc-partner-link>
    </rc-card-content>
  </rc-card>
  
</rc-partners-list>
`;

exports[`PartnersListComponent should create empty 1`] = `
<rc-partners-list>
  
</rc-partners-list>
`;
